python/ray/data/datasource/tfrecords_datasource.py

```diff
@@ -48,6 +74,25 @@ def _convert_example_to_dict(
     return record
 
 
+def _convert_arrow_to_examples(
+    arrow: "pyarrow.Table",
+) -> Iterable["tf.train.Example"]:
+    import tensorflow as tf
+
+    # Serialize each row[i] of the block to a tf.train.Example and yield it.
+    for i in range(arrow.num_rows):
```

Comment:
Supernit: An altnernative would be using the `table.to_pylist()` API, which converts the table to a list of record dicts:
```python
records = table.to_pylist()
for record in records:
    features = {key: _value_to_feature(value) for key, value in record.items()}
    yield tf.train.Example(features=tf.train.Features(feature=features))
```
Slightly simpler, at the cost of materializing the full table twice: once in the Arrow Table format, and once in the list of record dicts format... The fact that your method streams examples from the base table is a big pro, so keeping what you have is probably best, but just wanted to call out a slightly simpler alternative!

Reply:
Thanks suggestion @clarkzinzow. I also vote for @xcharleslin's current implementation, given the memory overhead.